hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/IMapSqlConnector.java

```diff
@@ -105,8 +106,12 @@ public List<MappingField> resolveAndValidateFields(
             @Nonnull NodeEngine nodeEngine,
             @Nonnull Map<String, String> options,
             @Nonnull List<MappingField> userFields,
-            @Nonnull String externalName
+            @Nonnull String[] externalName
     ) {
+        if (externalName.length > 1) {
```

Comment:
why? IMaps can have dots in names

Reply:
If that is the case, it needs to be quoted: `CREATE MAPPING .... EXTERNAL NAME \"map.with.dot\"`

Reply:
with backslashes? Currently we do not have schemas for IMaps but we possibly want to have. But even without them dots are natural way of grouping similar IMaps. Quoting adds friction when using them.

Reply:
Right, you don't need backslashes (sorry, copy-paste from Java code).

This doesn't change from the current situation, right now you also need to provide quotes if you have map with dot - `.` or also e.g. hyphen `-` in the name

Reply:
This actually works well:

```
CREATE MAPPING .... EXTERNAL NAME "schema.with.dot"."map.with.dot"
```

Reply:
but this: `CREATE MAPPING .... EXTERNAL NAME __sql.catalog` does not

Reply:
BTW, `CREATE MAPPING .... EXTERNAL NAME "schema.with.dot"."map.with.dot"` will not work for IMap because we allow only 1 component